nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -574,6 +582,27 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
     " milliseconds to smooth path." << std::endl;
 #endif
 
+  if (_debug_visualizations) {
+    // plot footprint path planned for debug
+    if (_planned_footprints_smoothed_publisher->get_subscription_count() > 0) {
+      auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+
+      visualization_msgs::msg::Marker clear_all_marker;
+      clear_all_marker.action = visualization_msgs::msg::Marker::DELETEALL;
+      marker_array->markers.push_back(clear_all_marker);
+
+      _planned_footprints_smoothed_publisher->publish(std::move(marker_array));
+
+      marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+      for (size_t i = 0; i < plan.poses.size(); i++) {
+        const std::vector<geometry_msgs::msg::Point> edge =
+          transformFootprintToEdges(plan.poses[i].pose, _costmap_ros->getRobotFootprint());
+        marker_array->markers.push_back(createMarker(edge, i, _global_frame, _clock->now()));
```

Comment:
Store the clock outside of the for loop and use here (so that all poses have the exact same timestamp)